homeassistant/components/tesla/climate.py

```diff
@@ -13,7 +13,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-SUPPORT_HVAC = [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+SUPPORT_HVAC = [HVAC_MODE_AUTO, HVAC_MODE_OFF]
```

Comment:
I think you want `HVAC_MODE_HEAT_COOL` which means it will heat or cool to a target.  `HVAC_MODE_AUTO` is more akin to a program schedule with no configurable set points:

https://github.com/home-assistant/home-assistant/blob/3b9f48b51dab8dc85fd71dbd0d081841c2fd12b7/homeassistant/components/climate/const.py#L12-L17

Reply:
Thanks. I can change it. The issue is there is no range available. Of course this was the argument in the original issue which was never resolved.

Reply:
The number of set-points is determined by the flags: `SUPPORT_TARGET_TEMPERATURE` and `SUPPORT_TARGET_TEMPERATURE_RANGE`...  What's the issue you're having with using `HEAT_COOL`?

Reply:
Correct, the issue is the comment doesn't clearly state HEAT_COOL allows for a range of 0. See this [comment](https://github.com/home-assistant/home-assistant/pull/28101#issuecomment-545579564). Again, this was discussed in the other thread. Happy to switch to HEAT_COOL.

Reply:
Does changing it to `HEAT_COOL` fix your issue?

Reply:
Yes.